### PR TITLE
Clean up LwM2M sample

### PR DESCRIPTION
**Pull request by SeppoTakalo**
### Simplify logging configuration
      
There is now only one log level configuration for application.
      
Also, there is now only couple of logging modules registered
with LOG_MODULE_REGISTER()
* app_lwm2m_client
* app_lwm2m
* app_sensors
* gnss_module

Multiple C files now share the same module and use
LOG_MODULE_DECLARE instead of each defining their own.

### Remove sensor avents
      
Simplify the sensor code by removing the applications events
that deliver sensor data. Instead write directly to the LwM2M
engine from the sensor worker. This is much more cleaner example.

Also remove unnecessary filtering of sensor data. It is enough
to filter with observation parameters.

Update timestamps on post_write callback.

### CODE REVIEW

**Review by juhaylinen — APPROVED** [review]

**Review by jarlamsa — APPROVED** [review]
LGTM

**Review by jheiskan81 — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/src/events/CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,4 @@
 zephyr_include_directories(include)
```
Comment (rlubos) [inline samples/nrf9160/lwm2m_client/src/events/CMakeLists.txt:7]: There's a typo in the commit title, `avents` -> `events`

**Review by rlubos — APPROVED** [review]
LGTM - please rebase the PR (conflict).

**Comment by SeppoTakalo** [comment]
Rebased and added a changelog and minor documentation update.
@peknis  please review.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -220,6 +220,10 @@ nRF9160 samples
     * Support for nRF7002 EK shield and Wi-Fi based location.
     * Location events and event handlers.
 
+  * Updated:
+
+    * Sensor module has been simplified, it does not use application events, filtering or configurable periods anymore.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:225]:
```suggestion
    * The sensor module has been simplified.
      It does not use application events, filtering, or configurable periods anymore.
```

File: samples/nrf9160/lwm2m_client/sample_description.rst [hunk 3]
```diff
@@ -185,16 +185,12 @@ For more information, see :ref:`notifications_setup_lwm2m`.
 Sensor Module
 =============
 
-The sample has a sensor module which, if enabled, reads the selected sensors, and updates the client's resource values if it detects a sufficiently large change in one of the values.
-The threshold for a sufficiently large change can be configured.
-For example, a change in temperature of one degree Celsius.
-
+The sample has a sensor module which, if enabled, reads the selected sensors, and updates the client's LwM2M resource values.
```
Comment (peknis) [inline samples/nrf9160/lwm2m_client/sample_description.rst:188]:
```suggestion
The sample has a sensor module which, if enabled, reads the selected sensors and updates the client's LwM2M resource values.
```

File: samples/nrf9160/lwm2m_client/sample_description.rst [hunk 4]
```diff
@@ -185,16 +185,12 @@ For more information, see :ref:`notifications_setup_lwm2m`.
...
 Each sensor can be enabled separately.
-The sampling period and change threshold of a sensor can also be configured independently of all the other sensors.
 
 The sensor module is intended to be used together with notifications.
-If notifications are enabled for a Sensor Value resource, and the corresponding sensor is enabled in the sensor module, a notification will be sent only when that value changes significantly (as specified by the change threshold).
-Thus, the bandwidth usage can be significantly limited, while simultaneously registering important changes in sensor values.
+If notifications are enabled for a Sensor Value resource, and the corresponding sensor is enabled in the sensor module, a notification will be sent when that value changes.
```
Comment (peknis) [inline samples/nrf9160/lwm2m_client/sample_description.rst:192]:
```suggestion
If notifications are enabled for a Sensor Value resource and the corresponding sensor is enabled in the sensor module, a notification will be sent when the value changes.
```

File: samples/nrf9160/lwm2m_client/sample_description.rst [hunk 5]
```diff
@@ -185,16 +185,12 @@ For more information, see :ref:`notifications_setup_lwm2m`.
...
+Frequency of notification packets are configured by LwM2M attributes set by the server.
```
Comment (peknis) [inline samples/nrf9160/lwm2m_client/sample_description.rst:193]:
```suggestion
The frequency of notification packets is configured by LwM2M attributes set by the server.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]

**Pull request merged**